### PR TITLE
[Dep] Update to php 8.1 requirement (manual)

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba this is split of https://github.com/rectorphp/rector-src/pull/1364 with manual change only with teskipped:

- rectify CI
- commented `MyCLabsClassToEnumRector` and `SpatieEnumClassToEnumRector` as it make `Enum` syntax upgrade needs to be skipped as it change `PhpVersion` class that extends : 

```diff
use MyCLabs\Enum\Enum;

- final class PhpVersion extends Enum
+ enum PhpVersion : int
```

which will need update currently manually of its usage:

```diff
-PhpVersion::PHP_52;
+PhpVersion::PHP_52->value;
```

and it seems cause change 

also other Enums, eg `ObjectReference` and `ApplicationPhase` manually:

```diff
-        if ($node->toString() === ObjectReference::PARENT()->getValue()) {
+        if ($node->toString() === ObjectReference::PARENT()->value) {
```

```diff
-        }, ApplicationPhase::PARSING());
+        }, ApplicationPhase::PARSING->value);
```

and we may need to create a downgrade rule for it to keep working on scoping.

 I commented the rules at https://github.com/rectorphp/rector-src/commit/99fa67bf060326680dcafc57ce8bc0e56b991d1e

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba all green 🎉 it is ready for review.

Note: the pending CI PHP 8.0 are probably cache or need you to disable it?

**Comment by TomasVotruba** [comment]
> Note: the pending CI PHP 8.0 are probably cache or need you to disable it?

It's because of requried names of jobs that must pass. Unfortunatelly Github sees thems as string, it's based on workflow name.
So changing string from "8.0" to "8.1" required manual change in settings. I'll handle it.

File: config/set/php81.php [hunk 1]
```diff
@@ -17,11 +17,11 @@
...
     $services = $containerConfigurator->services();
     $services->set(ReturnNeverTypeRector::class);
-    $services->set(MyCLabsClassToEnumRector::class);
+    //$services->set(MyCLabsClassToEnumRector::class);
     $services->set(MyCLabsMethodCallToEnumConstRector::class);
     $services->set(FinalizePublicClassConstantRector::class);
     $services->set(ReadOnlyPropertyRector::class);
-    $services->set(SpatieEnumClassToEnumRector::class);
+    //$services->set(SpatieEnumClassToEnumRector::class);
     $services->set(Php81ResourceReturnToObjectRector::class);
```
Comment (TomasVotruba) [inline config/set/php81.php]: These should be included in set so others can use them, but be skipped in our `rector.php`.

**Comment by TomasVotruba** [comment]
> which will need update currently manually of its usage:
>
>-PhpVersion::PHP_52;
>+PhpVersion::PHP_52->value;

Instead, the code using these enums should be upgraded from `string` to the Enum, e.g.:

```diff
interface MinPhpVersionInterface
{
-    public function provideMinPhpVersion(): int;
+    public function provideMinPhpVersion(): PhpVersion;
}

```

It should be handled in separated PR, maybe via a new Rector rule.

File: config/set/php81.php (hunk 1 above)
Comment (samsonasik) [reply config/set/php81.php]: updated https://github.com/rectorphp/rector-src/pull/1374/commits/0ee361c471aa92f06cf2847699ba81693aa2d676

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**